tests/integration/unhappy_path.py

```diff
@@ -0,0 +1,356 @@
+        name="_acme-dedicated", value="-"
+    )
+    cname: DnsZoneDomainReference = azure_dns_zone_manager.create_cname_record(
+        name="_acme-challenge." + dns_zone_record_name,
+        value=txt_record.name,
+    )
+
+    delegation_config = DnsZoneDomainReference(
+        dns_zone_resource_id=txt_record.dns_zone_resource_id,
+        name=cname.name.replace("_acme-challenge.", ""),
+    )
+
+    acme_config_manager.add_certificate_to_config(
+        cert_name=key_vault_cert_name,
+        domain_references=[delegation_config],
+    )
+
+    ## Test
+    client = AcmeDnsAzureClient(acme_config_manager.config)
+    results: [RotationResult] = client.issue_certificates()
+
+    ## Validate
+    for result in results:
+        assert result.result == CertbotResult.FAILED
+
+
+def test_create_cert_for_dns_delegation_shared_txt_shared_cert_with_minimum_permission_success(
+    acme_config_manager,
+    azure_key_vault_manager,
+    azure_dns_zone_manager,
+    azure_ad_manager,
+    config_file_path,
+    resource_name,
+    principal_id,
+):
+    ## Config
+    key_vault_cert_name = dns_zone_record_name = resource_name
+    config_file_path = config_file_path.replace(
+        "config.yaml", "no_permission_config.yaml"
+    )
+    acme_config_manager.base_config_from_file(file_path=config_file_path)
+
+    ## Init
+
+    txt_record: DnsZoneDomainReference = azure_dns_zone_manager.create_txt_record(
+        name="_acme-shared", value="-"
+    )
+    cname1: DnsZoneDomainReference = azure_dns_zone_manager.create_cname_record(
+        name="_acme-challenge." + dns_zone_record_name + "1",
+        value=txt_record.name,
+    )
+    cname2: DnsZoneDomainReference = azure_dns_zone_manager.create_cname_record(
+        name="_acme-challenge." + dns_zone_record_name + "2",
+        value=txt_record.name,
+    )
+
+    azure_ad_manager.create_role_assignment(
+        scope=txt_record.dns_zone_resource_id,
+        principal_id=principal_id,
+    )
+
+    delegation_config1 = DnsZoneDomainReference(
+        dns_zone_resource_id=txt_record.dns_zone_resource_id,
+        name=cname1.name.replace("_acme-challenge.", ""),
+    )
+    delegation_config2 = DnsZoneDomainReference(
+        dns_zone_resource_id=txt_record.dns_zone_resource_id,
+        name=cname2.name.replace("_acme-challenge.", ""),
+    )
+
+    acme_config_manager.add_certificate_to_config(
+        cert_name=key_vault_cert_name,
+        domain_references=[delegation_config1, delegation_config2],
+    )
+
+    ## Test
+    client = AcmeDnsAzureClient(acme_config_manager.config)
+    results: [RotationResult] = client.issue_certificates()
+
+    ## Validate
+    for result in results:
+        assert result.result == CertbotResult.CREATED
+    cn, san = azure_key_vault_manager.get_cn_and_san_from_certificate(
+        key_vault_cert_name
+    )
+    # TODO double check: what is the expected CN here? Is this correct: CN=testautohappypath1.{domain}
```

Comment:
Please check

Reply:
CN would be the the wirst domain. Second domain should be part of SAN